### PR TITLE
Geojson Validation

### PR DESCRIPTION
**Pull request by cKellyDesign**
Closes: #329 

This PR:
* installs @mapbox/geojsonhint
* validates geojson sourced layers
* blocks the layer from being rendered if there are non-"message" level errors

### CODE REVIEW

**Comment by cKellyDesign** [comment]
This is currently working as expected but it seems we get an error when compiling `gisida`.

Screen Shot 2019-10-25 at 11 32 13 AM

**Comment by cKellyDesign** [comment]
The error above is resolved by t[his suggestion to update the webpack configs](https://github.com/webpack-contrib/css-loader/issues/447#issuecomment-285598881).

File: src/map/prepareLayer.js [hunk 1]
```diff
@@ -270,7 +271,11 @@ function readData(mapId, layer, dispatch, doUpdateTsLayer) {
     if (layerObj.aggregate && layerObj.aggregate.filter) {
       layerObj.filterOptions = generateFilterOptions(layerObj);
     }
-    renderData(mapId, layerObj, dispatch, doUpdateTsLayer);
+    const geojsonErrors = hint(sourceURL.data).filter(e => e.level !== 'message');
+
+    if (!geojsonErrors || !geojsonErrors.length) {
+      renderData(mapId, layerObj, dispatch, doUpdateTsLayer);
+    }
```
Comment (kahummer) [inline src/map/prepareLayer.js]: This pr looks good. I'm thinking it would be awesome to have an error message shown when geojsonErrors.length is true

Comment (cKellyDesign) [reply src/map/prepareLayer.js]: @kahummer do we currently have a way to show such error messages?

Comment (kahummer) [reply src/map/prepareLayer.js]: @cKellyDesign how about logging the message on the console just to inform better what part of the geometry isn't correct. an else that logs the ``` geojsonErrors["message"] ```

Comment (cKellyDesign) [reply src/map/prepareLayer.js]: ok maybe we can do a `console.warn(..)` then

**Pull request merged**